helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -859,14 +866,24 @@ public boolean isGlobalRebalanceAsyncModeEnabled() {
 
   /**
    * Set the abnormal state resolver class map.
+   * @param resolverMap - the resolver map
+   *                    If null, the resolver map item will be removed from the config.
    */
   public void setAbnormalStateResolverMap(Map<String, String> resolverMap) {
-    if (resolverMap.values().stream()
-        .anyMatch(className -> className == null || className.isEmpty())) {
-      throw new IllegalArgumentException(
-          "Invalid Abnormal State Resolver Map definition. Class name cannot be empty.");
+    if (resolverMap == null) {
+      _record.getMapFields().remove(ClusterConfigProperty.ABNORMAL_STATES_RESOLVER_MAP.name());
+    } else {
+      if (resolverMap.entrySet().stream().anyMatch(e -> {
+        String stateModelDefName = e.getKey();
+        String className = e.getValue();
+        return stateModelDefName == null || stateModelDefName.isEmpty() || className == null
+            || className.isEmpty();
+      })) {
+        throw new IllegalArgumentException(
+            "Invalid Abnormal State Resolver Map definition. StateModel definition name and the resolver class name cannot be empty.");
```

Comment:
nit. 

StateModel definition name can't be null, resolver class name cannot be empty can't be null or empty.

Reply:
We have a test case for this?

Reply:
Yes, the test has been added. Let me modify the log string.